snipsmanager/commands/run.py

```diff
@@ -169,6 +170,44 @@ def handle_intent(self, intent, payload=None):
                 else:
                     getattr(skill, intent_def.action)()
 
+    def handle_dialogue_events_async(self, state):
+        """ Handle the dialogue API events."""
+        if(state == self.server.DIALOGUE_EVENT_STARTED):
+            state_name = "session_started"
+        elif(state == self.server.DIALOGUE_EVENT_ENDED):
+            state_name = "session_ended"
+        elif(state == self.server.DIALOGUE_EVENT_QUEUED):
+            state_name = "session_queued"
+
+        thread = threading.Thread(target=self.handle_dialogue_events, args=(state_name, ))
+        thread.start()
+
+    def handle_dialogue_events(self, name):
+        """ Handle the dialogue API events asynchronously."""
+
+        for skilldef in self.skilldefs:
+            dialogue_events_def = skilldef.find_dialogue_event(name)
+            if dialogue_events_def is None:
+                continue
+            if skilldef.package_name in self.skills:
+                skill = self.skills[skilldef.package_name]
+            elif skilldef.name in self.skills:
+                skill = self.skills[skilldef.name]
+            else:
+                continue
+            if dialogue_events_def.action.startswith("{%"):
+                # Replace variables in scope with random variables
+                # to prevent the skill from accessing/editing them.
+                action = dialogue_events_def.action \
+                    .replace("{%", "") \
+                    .replace("%}", "") \
+                    .replace("skilldef", "_snips_eejycfyrdfzilgfb") \
+                    .replace("intent_def", "_snips_jkqdruouzuahmgns")\
+                    .replace("snipsfile", "_snips_pdzdcpaygyjklngz") \
+                    .strip()
```

Comment:
@michaelfester Is it relevant to replace this variables in this context ? We added it because it's present also on the handle_notifications loop.